api/v1beta1/common_types.go

```diff
@@ -128,6 +128,16 @@ const (
 	// any changes to the actual object because it is a dry run) and the topology controller
 	// will receive the resulting object.
 	TopologyDryRunAnnotation = "topology.cluster.x-k8s.io/dry-run"
+
+	// ReplicasManagedByAnnotation is an annotation that indicates external (non-Cluster API) management of infra scaling.
+	// The practical effect of this is that the capi "replica" count should be passively derived from the number of observed infra machines,
+	// instead of being a source of truth for eventual consistency.
+	// This annotation can be used to inform MachinePool status during in-progress scaling scenarios.
+	ReplicasManagedByAnnotation = "cluster.x-k8s.io/replicas-managed-by"
+
+	// ExternalAutoscalerReplicasManagedByAnnotationValue is used with the "cluster.x-k8s.io/replicas-managed-by" annotation
+	// to indicate an external autoscaler enforces replica count.
+	ExternalAutoscalerReplicasManagedByAnnotationValue = "external-autoscaler"
```

Comment:
```suggestion
	ReplicasManagedByExternalAutoscalerAnnotationValue = "external-autoscaler"
```
nit: I think this should be the other way around. Looked at a few k/k "enum" values and they look e.g. like this: `PodFailurePolicyOnExitCodesOpIn`

Reply:
Okay looked a few more. They mixed pre- and suffix randomly.

Do we know if there is a canonical way to do this in Go or a precent in CAPI that we would want to follow for consistency?

Reply:
meh, it's just a const, no strong opinions, your suggestion seems fine to me

Reply:
Hm looked at some more of our constants. Looks like we follow more the suffix pattern (e.g. -Condition -Reason)

Let's keep your current const